widget/popup_menu.go

```diff
@@ -15,13 +15,47 @@ type PopUpMenu struct {
 	*Menu
 	canvas  fyne.Canvas
 	overlay *widget.OverlayContainer
+	Parent  fyne.CanvasObject
```

Comment:
This parent tracking seems like an anti-pattern - there is need for focus tracking across overlays, remembering the old one and re-instating it after popup removal would be cleaner.

Reply:
Yes, there should be a way of focus tracking across overlays. When that is implemented, the parent can be removed. (By the way, I have changed Parent to parent)

Reply:
In general it is preferable to put in place the correct implementation rather than workarounds waiting for complete solutions.
Of course every rule has an exception - but we'd want to know why this was required instead of fixing the underlying deficiency.